### PR TITLE
Handle undefined symbols at runtime

### PR DESCRIPTION
**Pull request by wsmoses**

### CODE REVIEW

File: src/compiler.jl [hunk 1]
```diff
@@ -908,6 +915,22 @@ function setfield_rev(B::LLVM.API.LLVMBuilderRef, OrigCI::LLVM.API.LLVMValueRef,
     return nothing
 end
 
+function get_binding_or_error_fwd(B::LLVM.API.LLVMBuilderRef, OrigCI::LLVM.API.LLVMValueRef, gutils::API.EnzymeGradientUtilsRef, normalR::Ptr{LLVM.API.LLVMValueRef}, shadowR::Ptr{LLVM.API.LLVMValueRef}, tapeR::Ptr{LLVM.API.LLVMValueRef})::Cvoid
+    CI = API.EnzymeGradientUtilsNewFromOriginal(gutils, OrigCI)
+    err = emit_error(LLVM.Builder(B), "Enzyme: unhandled forward for jl_get_binding_or_error")
```
Comment (vchuravy) [inline src/compiler.jl:920]: this seems wrong? We just want to call the original function.

**Review by vchuravy — CHANGES_REQUESTED** [review]

File: src/compiler.jl (hunk 1 above)
Comment (wsmoses) [reply src/compiler.jl:920]: We already call the original function [and don't remove that call], we just err after the fact, if for some reason resolution comes back ok

Comment (vchuravy) [inline src/compiler.jl:920]:
```suggestion
    err = emit_error(LLVM.Builder(B), "Enzyme: unreachable reached")
```

Comment (wsmoses) [reply src/compiler.jl:920]: So I wouldn't say this is necessarily an unreachable, so much as an unimplemented global, no?

**Review by vchuravy — APPROVED** [review]

**Pull request merged**